PyInstaller/building/build_main.py

```diff
@@ -218,6 +217,9 @@ def _get_module_collection_mode(mode_dict, name, noarchive=False):
     return mode_flags
 
 
+_default_cipher = object()
```

Comment:
Wouldn't it be better to have `_default_cipher = None`? This way, the existing spec files that do not use bytecode encryption (but due to current template, they set  `block_cipher = None` at the top and pass it via `cipher` argument) can continue working as-is? Or do we want to force everyone to purge all evidence that the cipher option ever existed?

Reply:
I was intending to go down the _force everyone to remove it_ route so that people aren't left with dead code in their spec files and we don't have to wait too long before we can remove these extra _if cipher is used: raise SystemExit(...)_ checks. Looking back at it now though it does feel a bit extreme. I'm guessing you also think this is too far?

I thought I might soften the blow a bit by changing the spec generation to remove the `cipher` logic if `--key` is not used. If that change was put in a separate commit which goes in a 5.x release (with the rest of this PR being v6.x only) then at least recently generated spec files would upgrade without complaint.